src/serde_xml_plist.rs

```diff
@@ -1,467 +1,657 @@
+            for (key, value) in self.0.iter() {
+                dict.serialize_field("key", key)?;
+                ValueHelper(value).serialize_within::<S>(&mut dict)?;
+            }
+            dict.end()
+        }
+    }
+
+    /// Serialises the inside of an array
+    struct ArrayInnerHelper<'lib>(&'lib [Value]);
+
+    impl Serialize for ArrayInnerHelper<'_> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            // e.g. `[Value::Integer(10), Value::String("hi")]` -> `struct { int: 10, string: hi }`
+            // -> `<int>10</int><string>hi</string>`
+            let mut array = serializer.serialize_struct("", self.0.len())?;
+            for value in self.0 {
+                ValueHelper(value).serialize_within::<S>(&mut array)?;
+            }
+            array.end()
+        }
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[derive(Deserialize)]
+    #[derive(Serialize, Deserialize)]
     struct TestMe {
-        #[serde(deserialize_with = "deserialize_dict")]
+        #[serde(with = "super")]
         lib: Dictionary,
     }
 
-    #[test]
-    fn deserialize_everything() {
-        let xml = r#"
-<?xml version='1.0' encoding='UTF-8'?>
-  <object>
-  <lib>
-    <dict>
-      <key>hasLoadedLib</key>
-      <string>Absolutely!</string>
-      <key>anArray</key>
-      <array>
-        <dict>
-          <key>class</key>
-          <string>aristocracy</string>
-          <key>heft</key>
-          <real>42.42</real>
-        </dict>
-        <integer>6</integer>
-      </array>
-      <key>isWorking</key>
-      <true/>
-      <key>isBroken</key>
-      <false/>
-      <key>bestBefore</key>
-      <date>2345-01-24T23:22:21Z</date>
-      <key>payload</key>
-      <data>
-      dSBnb3QgMHduZWQ=
-      </data>
-    </dict>
-  </lib>
-  </object>
-"#;
-
-        let readme: TestMe = quick_xml::de::from_str(xml).unwrap();
-        assert_eq!(readme.lib.get("hasLoadedLib").unwrap().as_string(), Some("Absolutely!"));
-        let array = readme.lib.get("anArray").unwrap().as_array().unwrap();
-        assert_eq!(
-            array[0].as_dictionary().and_then(|d| d.get("class")),
-            Some(&Value::String("aristocracy".into()))
-        );
-        assert_eq!(array[0].as_dictionary().and_then(|d| d.get("heft")), Some(&Value::Real(42.42)));
-        assert_eq!(array[1].as_signed_integer(), Some(6));
-        assert_eq!(readme.lib.get("isWorking"), Some(&Value::Boolean(true)));
-        assert_eq!(readme.lib.get("isBroken"), Some(&Value::Boolean(false)));
-        assert_eq!(
-            readme.lib.get("bestBefore").and_then(Value::as_date).map(|d| d.to_xml_format()),
-            Some("2345-01-24T23:22:21Z".into())
-        );
-        assert_eq!(
-            readme.lib.get("payload").and_then(Value::as_data),
-            Some("u got 0wned".as_bytes())
-        );
-    }
+    mod de {
```

Comment:
Is there any reason to have separate submodules in the tests? I'd just put all the tests in `test` unless there's some specific rationale.